awx/api/serializers.py

```diff
@@ -2515,17 +2545,27 @@ def validate_extra_vars(self, value):
 
     def get_summary_fields(self, obj):
         summary_fields = super(JobTemplateSerializer, self).get_summary_fields(obj)
-        if 'pk' in self.context['view'].kwargs and self.version > 1:  # TODO: remove version check in 3.3
+        if self.is_detail_view:
+            all_creds = []
             extra_creds = []
-            for cred in obj.extra_credentials.all():
-                extra_creds.append({
+            for cred in obj.credentials.all():
```

Comment:
`.select_related('credential_type')`?

Reply:
can't you alternatively do `.select_related('credential_type__kind')`? That could limit the volume of the data from the database appropriately for summary_fields.

Reply:
`credential_type__kind` isn't a foreign key field, so that wouldn't work.